src/abstracts/SablierV2Lockup.sol

```diff
@@ -246,19 +246,13 @@ abstract contract SablierV2Lockup is
             revert Errors.SablierV2Lockup_StreamDepleted(streamId);
         }
 
-        bool isCallerStreamSender = _isCallerStreamSender(streamId);
-
-        // Checks: `msg.sender` is the stream's sender, the stream's recipient, or an approved third party.
-        if (!isCallerStreamSender && !_isCallerStreamRecipientOrApproved(streamId)) {
-            revert Errors.SablierV2Lockup_Unauthorized(streamId, msg.sender);
-        }
-
         // Retrieve the recipient from storage.
         address recipient = _ownerOf(streamId);
 
-        // Checks: if `msg.sender` is the stream's sender, the withdrawal address must be the recipient.
-        if (isCallerStreamSender && to != recipient) {
-            revert Errors.SablierV2Lockup_InvalidSenderWithdrawal(streamId, msg.sender, to);
+        // Checks: if `to` is not the recipient, then `msg.sender` must be the stream's recipient or an approved third
+        // party.
+        if (to != recipient && !_isCallerStreamRecipientOrApproved(streamId)) {
```

Comment:
I intentionally used that order.

I think, in most of the cases, `to` will be the recipient. Now since solidity has short-circuiting for boolean operators, it is more gas efficient to check `to` first (as `to != recipient` will return `false` and then it won't check the second condition). While `_isCallerStreamRecipientOrApproved ` requires minimum 1 storage access.

And if we reverse the order, it will access storage every single time.

Reply:
ohh, i see, this is correct. it is better to keep this order

should we update the comment the custom error name as i suggest?

Reply:
> should we update the comment the custom error name as i suggest?

Sorry I didn't get this.

Reply:
Yes, it looks better.